library/core/src/main/java/com/google/android/exoplayer2/audio/SonicAudioProcessor.java

```diff
@@ -129,7 +138,7 @@ public boolean configure(int sampleRateHz, int channelCount, @Encoding int encod
 
   @Override
   public boolean isActive() {
-    return Math.abs(speed - 1f) >= CLOSE_THRESHOLD || Math.abs(pitch - 1f) >= CLOSE_THRESHOLD;
+    return isActive && (Math.abs(speed - 1f) >= CLOSE_THRESHOLD || Math.abs(pitch - 1f) >= CLOSE_THRESHOLD);
```

Comment:
`AudioTrack.setPlaybackParameters` currently assumes that it's possible to change the playback parameters unless it's using passthrough output. When using `ENCODING_PCM_FLOAT` it will behave as if the parameters were applied, but this processor will actually be inactive in that case so the parameters won't be applied.

Thinking about `AudioProcessor`s more generally in the context of this change: if not in passthrough mode, at the moment `AudioTrack` always applies `ResamplingAudioProcessor` first to get audio into 16-bit PCM format then applies the other processors in turn, all of which expect 16-bit PCM input. In the current design, it's expected that audio processors other than the resampler should input and output PCM 16-bit encoding, and they will throw if they get something else.

If the main use case for this change is to decode to 32-bit float directly from `FfmpegAudioRenderer`, I wonder if we can simplify things by making `ENCODING_PCM_FLOAT` work like passthrough, at least initially. (Perhaps with a new boolean `encodedOutput` in `AudioTrack`, which is always true if `passthrough` is true but also true if writing float data.) Then the audio processors don't need to be modified. Then in later changes we could add support for resampling 24/32-bit PCM to float as a special case, then potentially handle float input in the other audio processors if there is enough demand to justify the added complexity. What do you think?